alf/algorithms/mi_estimator.py

```diff
@@ -72,6 +86,11 @@ class MIEstimator(Algorithm):
     assumption that y samples from one batch are independent. If the additional
     memory is not a concern, we recommend 'buffer' sampler so that there is no
     need to worry about the assumption of independence.
+
+    MIEstimator can be also used to estimate conditional mutual information
+    MI(X,Y|Z). In this case, you should let `x` to represent X and Z, and `y` to
```

Comment:
This statement seems incorrect, as MI(X, Y | Z) is not equal to MI([X,Z], Y). Because 'x' and 'y' in the former both condition on 'z' in the denominator while only one of them conditions on 'z' in the latter.  This is same for both 'JSD' and 'ML'

Reply:
The way it is used make it estimating MI(X, Y|Z). For MI([X,Z], Y), y_distritution should sample the marginal distribution P(Y). For MI(X,Y|Z), y_distribution should sample P(Y|z)

Reply:
ok I see.